File: src/main/java/gr/apt/persistence/entity/superclass/AbstractEntity.java
package gr.apt.persistence.entity.superclass;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import io.quarkus.runtime.annotations.RegisterForReflection;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Basic;
import javax.persistence.Column;
import javax.persistence.MappedSuperclass;
import javax.persistence.PrePersist;
import java.io.Serializable;
import java.math.BigInteger;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@MappedSuperclass
@RegisterForReflection
public class AbstractEntity extends PanacheEntityBase implements Serializable {

    @Basic
    @Column(name = "created")
    private LocalDateTime created;
    @Basic
    @Column(name = "created_by")
    private BigInteger createdBy;
    @Basic
    @Column(name = "updated")
    private LocalDateTime updated;
    @Basic
    @Column(name = "updated_by")
    private BigInteger updatedBy;

    @PrePersist
    private void prePersist() {
        if (created == null) {
            created = LocalDateTime.now();
            createdBy = BigInteger.ONE;
        }
        updated = LocalDateTime.now();
        updatedBy = BigInteger.ONE;
    }
}

File: src/main/java/gr/apt/persistence/entity/superclass/AbstractEntity.kt
package gr.apt.persistence.entity.superclass

import io.quarkus.hibernate.orm.panache.PanacheEntityBase
import io.quarkus.runtime.annotations.RegisterForReflection
import java.io.Serializable
import java.math.BigInteger
import java.time.LocalDateTime
import javax.persistence.*

@MappedSuperclass
@RegisterForReflection
open class AbstractEntity : PanacheEntityBase(), Serializable {
    @Basic
    @Column(name = "created")
    open var created: LocalDateTime? = null

    @Basic
    @Column(name = "created_by")
    open var createdBy: BigInteger? = null

    @Basic
    @Column(name = "updated")
    open var updated: LocalDateTime? = null

    @Basic
    @Column(name = "updated_by")
    open var updatedBy: BigInteger? = null

    @PrePersist
    private fun prePersist() {
        if (created == null) {
            created = LocalDateTime.now()
            createdBy = BigInteger.ONE
        }
        updated = LocalDateTime.now()
        updatedBy = BigInteger.ONE
    }

    @PreUpdate
    private fun preUpdate(){
        updated = LocalDateTime.now()
        updatedBy = BigInteger.ONE
    }
}